pkg/scheduler/preemption/preemption_test.go

```diff
@@ -43,26 +43,34 @@ import (
 	"sigs.k8s.io/kueue/pkg/constants"
 	controllerconstants "sigs.k8s.io/kueue/pkg/controller/constants"
 	"sigs.k8s.io/kueue/pkg/features"
-	"sigs.k8s.io/kueue/pkg/hierarchy"
 	"sigs.k8s.io/kueue/pkg/scheduler/flavorassigner"
 	"sigs.k8s.io/kueue/pkg/util/slices"
 	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
 	"sigs.k8s.io/kueue/pkg/workload"
 )
 
-var snapCmpOpts = []cmp.Option{
-	cmpopts.EquateEmpty(),
-	cmpopts.IgnoreUnexported(hierarchy.Cohort[*cache.ClusterQueueSnapshot, *cache.CohortSnapshot]{}),
-	cmpopts.IgnoreUnexported(hierarchy.ClusterQueue[*cache.CohortSnapshot]{}),
-	cmpopts.IgnoreUnexported(hierarchy.Manager[*cache.ClusterQueueSnapshot, *cache.CohortSnapshot]{}),
-	cmpopts.IgnoreFields(cache.ClusterQueueSnapshot{}, "AllocatableResourceGeneration"),
-	cmp.Transformer("Cohort.Members", func(s sets.Set[*cache.ClusterQueueSnapshot]) sets.Set[kueue.ClusterQueueReference] {
-		result := make(sets.Set[kueue.ClusterQueueReference], len(s))
-		for cq := range s {
-			result.Insert(cq.Name)
+type nodeKey struct {
+	cohort bool
+	name   string
+}
+
+func resourceNodes(snapshot *cache.Snapshot) map[nodeKey]cache.ResourceNode {
+	nodes := map[nodeKey]cache.ResourceNode{}
+	for _, cohort := range snapshot.Cohorts() {
+		nodes[nodeKey{cohort: true, name: string(cohort.Name)}] = cohort.ResourceNode
+	}
+	for _, cq := range snapshot.ClusterQueues() {
+		nodes[nodeKey{cohort: false, name: string(cq.Name)}] = cq.ResourceNode
+	}
+	// clear zeros
```

Comment:
Interesting, can you clarify why we do so? Please expand the comment. 

Would it help to add a cmpopt.EquateEmpty to cmp.Diff?

Reply:
EquateEmpty didn't work. I got IgnoreMapEntries to work